clients/python-apiserver-client/python_apiserver_client/params/headnode.py

```diff
@@ -0,0 +1,126 @@
+from .volumes import *
+from .environmentvariables import *
+
+DEFAULT_HEAD_START_PARAMS = {"dashboard-host": "0.0.0.0", "metrics-export-port": "8080"}
+
+
+class ServiceType(enum.Enum):
+    ClusterIP = "ClusterIP"
+    NodePort = "NodePort"
+    LoadBalancer = "LoadBalancer"
+
+
+class HeadNodeSpec:
+    """
+    HeadNodeSpec is used to define Ray cluster head node configuration.
+    It provides APIs to create, stringify and convert to dict.
+
+    Methods:
+    - Create head node specification: gets the following parameters:
+        compute_template - required, the computeTemplate of head node group
+        ray_start_params - required, Ray start parameters
+        image - optional, image used for head node
+        service_type - optional (ServiceType), service type foe headnode
+        enable_ingress - optional, allow to enable ingress for dashboard
+        volumes - optional, a list of volumes to attach to head node
+        service_account - optional, a service account (has to exist) to run head node
+        image_pull_secret - optional, secret to pull head node image from registry
+        environment - optional, environment variables for head pod
+        annotations - optional, annotations for head node
+        labels - optional, labels for head node
+    """
+    def __init__(self, compute_template: str, ray_start_params: dict[str, str], image: str = None,
+                 service_type: ServiceType = None, enable_ingress: bool = False,
+                 volumes: list[BaseVolume] = None, service_account: str = None, image_pull_secret: str = None,
+                 environment: EnvironmentVariables = None, annotations: dict[str, str] = None,
+                 labels: dict[str, str] = None) -> None:
+
+        self.compute_template = compute_template
+        self.ray_start_params = ray_start_params
+        self.ray_start_params.update(DEFAULT_HEAD_START_PARAMS)
+        self.image = image
+        self.service_type = service_type
+        self.enable_ingress = enable_ingress
+        self.volumes = volumes
+        self.service_account = service_account
+        self.image_pull_secret = image_pull_secret
+        self.environment = environment
+        self.annotations = annotations
+        self.labels = labels
+
+    def to_string(self) -> str:
```

Comment:
Wouldn't it be simpler to use something like pydantic for these dataclasses? Then you would get the to_string and to_dict methods for free.

Reply:
If all the transformations you need are between snake case and camel case it's not that much code: https://medium.com/analytics-vidhya/camel-case-models-with-fast-api-and-pydantic-5a8acb6c0eee

Reply:
Lots of boilerplate here...